tests/x11/yast2_lan_restart.pm

```diff
@@ -42,7 +42,20 @@ sub check_network {
     $status //= 'no_restart';
     wait_still_screen;
     send_key 'alt-o';                     # OK
-    assert_screen 'yast2_closed_xterm_visible', 120;
+                                          # new: warning pops up for firewall, alt-y for assign it to zone
+    assert_screen([qw(yast2-lan-restart_firewall_active_warning yast2_closed_xterm_visible)], 120);
+    if (match_has_tag 'yast2-lan-restart_firewall_active_warning') {
+        send_key 'alt-y';
+        wait_still_screen;
+        send_key 'alt-n';
+        wait_still_screen;
+        send_key 'alt-o';
+    }
+    else {
+        assert_screen 'yast2_closed_xterm_visible', 120;
+    }
+    #  for sure that it comes back to root console
+    wait_still_screen;
```

Comment:
The above `assert_screen 'yast2_closed_xterm_visible', 120;` outside of the if-else should do that job. I don't see the need for this `wait_still_screen`

Reply:
@okurz it's more for my personal knowledge, but could you explain me why the `assert_screen 'yast2_closed_xterm_visible', 120;` after the `else` is needed? The `assert_screen ([qw(yast2-lan-restart_firewall_active_warning yast2_closed_xterm_visible)], 120);1` should not do the job already?

I don't want to insist ;-) I just want to understand :)